website/docs/language/import/generating-configuration.mdx

```diff
@@ -0,0 +1,137 @@
+---
+page_title: Import - Generating Configuration
+description: >-
+  Generate configuration and manage existing resources with Terraform using configuration-driven import.
+---
+
+# Generating Configuration
+
+~> **Experimental:** Configuration generation is available in Terraform v1.5 as an experimental feature. Later minor versions may contain changes to the formatting of generated configuration and behavior of the `terraform plan` command using the `-generate-config-out` flag.
+
+Terraform can generate HCL for resources that do not already exist in your configuration. Terraform produces HCL to act as a template that contains Terraform's best guess at the appropriate value for each resource argument. 
+
+Starting with Terraform's generated HCL, we recommend iterating to find your ideal configuration by removing some attributes, adjusting the value of others, and rearranging `resource` blocks into files and modules as appropriate.
+
+To generate configuration, use `terraform plan` with the `-generate-config-out` flag and supply a new file path. Do not supply a path to an existing file, this command creates a new one. 
+
+```
+terraform plan -generate-config-out=generated_resources.tf
```

Comment:
Suggesting adding the`$`, I think it's probably fine either way, but it might look nicer with it:
```suggestion
$ terraform plan -generate-config-out=generated_resources.tf
```

Reply:
Another potential benefit, assuming docs get rendered in the same way as tutorials, is that the leading `$` within a triple-tick-block seems to trigger color coding (red dollar sign, quoted strings in yellow, etc).